src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -996,83 +999,78 @@ class npc_dkc1_gothik : public CreatureScript
 
 };
 
-class npc_scarlet_ghoul : public CreatureScript
+struct npc_scarlet_ghoul : public ScriptedAI
 {
-public:
-    npc_scarlet_ghoul() : CreatureScript("npc_scarlet_ghoul") { }
+    npc_scarlet_ghoul(Creature* creature) : ScriptedAI(creature)
+    {
+        Initialize();
+    }
 
-    CreatureAI* GetAI(Creature* creature) const override
+    void Initialize()
     {
-        return new npc_scarlet_ghoulAI(creature);
+        if (urand(0, 1))
+            if (Unit* owner = me->GetOwner())
+                Talk(SAY_LINE_0, owner);
+
+        me->SetReactState(REACT_DEFENSIVE);
     }
 
-    struct npc_scarlet_ghoulAI : public ScriptedAI
+    void FindMinions(Unit* owner)
     {
-        npc_scarlet_ghoulAI(Creature* creature) : ScriptedAI(creature)
-        {
-            // Ghouls should display their Birth Animation
-            // Crawling out of the ground
-            //DoCast(me, 35177, true);
-            //me->MonsterSay("Mommy?", LANG_UNIVERSAL, 0);
-            me->SetReactState(REACT_DEFENSIVE);
-        }
+        std::list<Creature*> MinionList;
+        owner->GetAllMinionsByEntry(MinionList, NPC_GHOULS);
 
-        void FindMinions(Unit* owner)
+        if (!MinionList.empty())
         {
-            std::list<Creature*> MinionList;
-            owner->GetAllMinionsByEntry(MinionList, NPC_GHOULS);
-
-            if (!MinionList.empty())
+            for (std::list<Creature*>::const_iterator itr = MinionList.begin();
```

Comment:
Please keep this on 1 line

Reply:
Or we modernize it while at it and go with `for (Creature* creature : MinionList)`